lib/miq_pglogical.rb

```diff
@@ -69,6 +71,19 @@ def self.save_global_region(subscriptions_to_save, subscriptions_to_remove)
 
   private
 
+  def with_connection_error_handling
+    retry_attempted = false
+    yield
+  rescue PG::ConnectionBad => bad_connection_err
+    if retry_attempted
+      raise bad_connection_err
+    else
+      pglogical(true)
+      retry_attempted = true
+      retry
+    end
+  end
```

Comment:
Can this just refer to the other duplicate block in pglogical_subscription (or vice versa?).  I feel like we should avoid the duplicated code, and I don't really have a preference in which module it belongs.

Reply:
@Fryguy yeah, I was thinking of doing the same thing, but wasn't sure where it was best to live, and figured as a first pass, it was better to have it isolated instead of coupling these two models together.

That said, I don't mind trying this up, so I can put something together to do that.